src/test/resources/conditionalizePermissions/let.vpr

```diff
@@ -0,0 +1,15 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+field f: Int
+
+function ok(): Bool
+
+
+function val(): Ref
+    requires ok()
+
+method test()
+{
+    inhale ok() ==> let x == (val()) in acc(x.f)
+}
```

Comment:
is this test checking that we can push the perm amount to an access predicate "inside" a let expression? If so, would there be a way to adapt this test to fail if that is not the case?

Reply:
This test is supposed to ensure that we *don't* push the ``ok()`` condition too far in. If we push it into the let-expression, i.e., to get ``inhale let x == (val()) in acc(x.f, ok() ? write : none)``, then ``val()`` isn't well-defined, and the test would fail.,

Reply:
Can you please add this as a comment to the test, please?

Reply:
Yes, that makes sense, done.